Xamarin.Forms.Core/Shell/ShellItem.cs

```diff
@@ -21,6 +21,20 @@ public FlyoutItem()
 		{
 
 		}
+
+		public static readonly new BindableProperty IsVisibleProperty =
```

Comment:
in Shell code, we often reuse the same BP for attached and non-attached scenarios. I think this applies here too

Reply:
overriding the BP will likely confuse users, and eventually tooling

Reply:
The main reason for this is the ability to articulate different types of visibility. Maybe I need a new property?

For example someone could do


```XAML
<FlyoutItem />
<FlyoutItem FlyoutDisplayOptions="AsMultipleItems" >
      <TabBar FlyoutItem.IsVisible="false".  <!-- this means just don't show up in the flyout -->
      <TabBar IsVisible="false"  <!-- this means just don't show up anywhere -->
</FlyoutItem>
<FlyoutItem FlyoutDisplayOptions="AsMultipleItems" IsVisible="false">  <!-- this means just don't show up anywhere -->
</FlyoutItem>

```

So the idea was to use the different attached properties to articulate what the `IsVisible` is visually being applied to

Does it need to be an enum or a different BP?

Reply:
@StephaneDelcroix are you aa bigger fan of having something like

`Shell.FlyoutitemIsVisible`

Reply:
no, that's good. make sure the XAML is parsed as intended